core/src/main/scala/com/banno/cosmos4s/BulkOps.scala

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright 2020 Jack Henry & Associates, Inc.®
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.banno.cosmos4s
+
+import cats._
+import cats.effect._
+import cats.syntax.all._
+import io.circe.Json
+import com.microsoft.azure.documentdb.{DocumentClient, PartitionKeyDefinition}
+import com.microsoft.azure.documentdb.bulkexecutor.{BulkImportResponse, DocumentBulkExecutor}
+
+trait BulkOps[F[_], V] {
+  def insert(value: List[V]): F[Unit]
+  def upsert(value: List[V]): F[Unit]
+
+  def mapK[G[_]](fk: F ~> G): BulkOps[G, V] = new BulkOps.MapKBulkOps[F, G, V](this, fk)
+  def contramapValue[A](f: A => V): BulkOps[F, A] = new BulkOps.ContramapValue[F, A, V](this, f)
+}
+
+object BulkOps {
+
+  /**
+   * https://docs.microsoft.com/en-us/azure/cosmos-db/bulk-executor-java
+   * It is recommended to instantiate a single DocumentBulkExecutor object for the entire application
+   * within a single virtual machine that corresponds to a specific Azure Cosmos container.
+   */
+  def impl[F[_]: Sync](
+      client: DocumentClient,
+      database: String,
+      collection: String,
+      partitionKey: PartitionKeyDefinition,
+      offerThroughput: Int,
+      maxConcurrencyPerPartitionRange: Int
+  ): Resource[F, BulkOps[F, Json]] =
+    Resource
```

Comment:
Why a resource at all if you don't have a shutdown?

Reply:
just to give the semantic of having one instance

Reply:
``public class DocumentBulkExecutor implements AutoCloseable``

Reply:
https://docs.microsoft.com/en-us/azure/cosmos-db/bulk-executor-java